.github/workflows/build-pr.yml

```diff
@@ -0,0 +1,64 @@
+name: Build-PR-test
+
+on:
+  # run in a read-write copy of the target of the PR (CombatExtended-Continued/CombatExtended/Development)
+  pull_request_target: 
+    branches: [ Development ]
+
+jobs:
+  build:
+    # Could use windows with dotnet, but I don't know the tooling.
+    # Also, mono is the less advanced compiler, so if it builds successfully here, it'll work anywhere.
+    runs-on: ubuntu-latest 
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: prepare rebuild
+        run: |
+          # Set up who gets blamed for the added assembly
+          git config --global user.email "actions@github.com"
+          git config --global user.name "GitHub Actions"
+
+          # Grab the code from the PR, using the version from the PR in case of conflicts.
+          git pull --allow-unrelated-histories -X theirs origin ${{ github.event.pull_request.head.sha }}:
+
+          # Grab the RimWorld reference assemblies from nuget
+          wget https://www.nuget.org/api/v2/package/Krafs.Rimworld.Ref/1.2.2753 -O rwref.zip
```

Comment:
Is there an endpoint to grab `latest`? Or perhaps read from `csproj` to make sure it compiles against the same version from local dev?

Reply:
On that note, why not just use `dotnet restore` or `nuget restore`?

Reply:
There is no dynamic link to the latest version of a nuget package.  I could either grab the version from the `csproj`, or scrape the index page off `nugget.org` and grab the latest version.  Normally, this will only matter when RimWorld comes out with a new release.  Which approach we take depends on how we want to handle new RW releases.

If we want to always build CE against the latest RW version, and can tolerate our build system breaking when a new RW version drops (and the nugget package updates), then grabbing the latest version from nugget makes sense.  If we want to manually set it, then either reading the `csproj` from `Development`, or just statically setting the package version is the way to go.  The package version could also get read from the `csproj` in the PR, which would let us test new versions without making changes to `Development`.  I didn't do that here because the current `csproj` doesn't include a package reference.  We could add one, but I think that should be discussed in its own PR.

Reply:
> On that note, why not just use `dotnet restore` or `nuget restore`?

As for `dotnet restore` or similar, I am not familiar enough with the tooling to be certain of the security implications of doing that.  This action runs on a read/write checkout of the `CombatExtended-Continued/CombatExtended` repository, into which arbitrary code has been copied by an untrusted third party.  In the execution environment are *secrets* which must not be divulged to the third party.  Under no circumstances is it safe to *run* code from the PR.  (Note that the action will fire before any of us have a chance to review the PR).  This is why `Make.py` gets pulled from the Development branch directly.  `mcs` also doesn't execute any of the files it compiles.  I know it is possible to include post-compile hooks in `.csproj`, but I am uncertain of the limits of that.  If it is possible to define a `post-restore` hook (or any other hook which triggers on the `restore` subcommand), that would preclude running those commands on the PR-version of the code.  We could get around that by pulling a fresh `.csproj` from the main branch, but that precludes testing changes to it via this system.

While slightly more brittle, the xml-walking dumbs-as-rocks build script is a much less general, much smaller attack surface for a nefarious PR.  (Not that I think someone is likely to target CE specifically, but it is conceivable that someone may notice we have actions enabled and treat it as a target of opportunity).